### DISCUSSION TITLE
docker compose 无法启动一直报端口被占用

### DISCUSSION THREAD

**Discussion by ximengyi** [apache/apisix#8704] [Q&A]
```
/usr/local/openresty//luajit/bin/luajit ./apisix/cli/apisix.lua init_etcd
2023/01/17 20:11:03 [emerg] 1#1: bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
nginx: [emerg] bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
2023/01/17 20:11:03 [emerg] 1#1: bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
nginx: [emerg] bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
2023/01/17 20:11:03 [emerg] 1#1: bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
nginx: [emerg] bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
2023/01/17 20:11:03 [emerg] 1#1: bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
nginx: [emerg] bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
2023/01/17 20:11:03 [emerg] 1#1: bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
nginx: [emerg] bind() to unix:/usr/local/apisix/conf/config_listen.sock failed (98: Address already in use)
2023/01/17 20:11:03 [emerg] 1#1: still could not bind()
nginx: [emerg] still could not bind()
````

请问这个错误是哪个端口被占用了，在哪里修改端口？
Which port is occupied by this error and where to modify the port？

**Comment by soulbird** [comment]
Check if this file /usr/local/apisix/conf/config_listen.sock exists, and delete it if it exists

**Reply by ximengyi** [reply]
/usr/local/apisix/conf/config_listen.sock is not exists .
Finally, I solved this problem by  force deleting container docker rm  docker rm -f docker-apisix_apisix_1
and then docker-compose -p docker-apisix up -d

**Comment by spatxos** [comment]
When the server is restarted, this problem still exists, is there a better solution?

**Reply by ximengyi** [reply]
i found a better solution 
change your docker-compose.yml
find  lines
depends_on:
        - etcd
comment out
##depends_on:
\#      - etcd
and then
docker-compose up -d apisix

**Comment by flylan** [comment]
I have also encountered the same problem, and my solution is to rewrite the entrypoint of the Apisix image. Although this is not a good suggestion, it is currently a good solution

**docker-compose.yaml**
version: "3.7"

services:
  apisix:
    container_name: apisix
    image: apisix:3.8.0-debian
    restart: always
    entrypoint: /entrypoint.sh
    volumes:
        - "./entrypoint.sh:/entrypoint.sh:ro"


**entrypoint.sh**

#!/bin/bash
set -e

if [ -S "/usr/local/apisix/conf/config_listen.sock" ]; then
  rm -f /usr/local/apisix/conf/config_listen.sock
fi

/docker-entrypoint.sh docker-start
